varnish-operator/templates/_helpers.tpl

```diff
@@ -0,0 +1,19 @@
+{{/*
+Return the proper Varnish Operator image name
+*/}}
+{{- define "varnish-operator.image" -}}
+    {{- $registryName := .Values.container.registry -}}
+    {{- $repositoryName := .Values.container.repository -}}
+    {{- $separator := ":" -}}
+    {{- $termination := .Values.container.tag | toString -}}
+    {{- if .Values.container.digest }}
+        {{- $separator = "@" -}}
+        {{- $termination = .Values.container.digest | toString -}}
+    {{- end -}}
+    {{- if .Values.container.image }}
+        {{- $image = .Values.container.image -}}
+        {{- printf "%s" $image -}}
+    {{- else -}}
+        {{- printf "%s/%s%s%s" $registryName $repositoryName $separator $termination -}}
+    {{- end -}}
```

Comment:
If you are going to go that route, then I would order the code like this to make it both obvious what you are doing AND to avoid getting values that you later ignore. This will make it easier for people to read and understand.

```suggestion
    {{- if .Values.container.image }}
        {{- $image = .Values.container.image -}}
        {{- printf "%s" $image -}}
    {{- else -}}
        {{- $registryName := .Values.container.registry -}}
        {{- $repositoryName := .Values.container.repository -}}
        {{- $separator := ":" -}}
        {{- $termination := .Values.container.tag | toString -}}
        {{- if .Values.container.digest }}
            {{- $separator = "@" -}}
            {{- $termination = .Values.container.digest | toString -}}
        {{- end -}}
        {{- printf "%s/%s%s%s" $registryName $repositoryName $separator $termination -}}
    {{- end -}}
```

Reply:
You are right it is way better in this order :+1: